cmd/secretcli/secret20.go

```diff
@@ -0,0 +1,371 @@
+		Args:  cobra.MinimumNArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			contractAddrStr, err := addressFromBechOrLabel(args[0], cliCtx)
+			if err != nil {
+				return err
+			}
+
+			contractAddr, err := sdk.AccAddressFromBech32(contractAddrStr)
+			if err != nil {
+				return errors.New("invalid contract address or label")
+			}
+
+			toAddr, err := sdk.AccAddressFromBech32(args[1])
+			if err != nil {
+				return errors.New("invalid recipient address")
+			}
+
+			amount := args[2]
+			_, err = strconv.ParseUint(amount, 10, 64)
+			if err == nil {
+				return errors.New("invalid amount format")
+			}
+
+			msg := sendCoinMsg(toAddr, amount)
+
+			return cli.ExecuteWithData(cmd, contractAddr, msg, "", false, "", "", cdc)
+		},
+	}
+
+	return cmd
+}
+
+func s20CreatingViewingKey(cdc *codec.Codec) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "create-viewing-key [contract address or label]",
+		Short: "Create a new viewing key. To view the resulting key, use 'secretcli q compute tx <TX_HASH>'",
+		Long: `This allows a user to generate a key that enables off-chain queries. 
+This way you can perform balance and transaction history queries without waiting for a transaction on-chain. 
+This transaction will be expensive, so you must have about 3,000,000 gas in your account to perform this step.
+ This is intended to make queries take a long time to execute to be resistant to brute-force attacks.`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			contractAddrStr, err := addressFromBechOrLabel(args[0], cliCtx)
+			if err != nil {
+				return err
+			}
+
+			contractAddr, err := sdk.AccAddressFromBech32(contractAddrStr)
+			if err != nil {
+				return errors.New("invalid contract address or label")
+			}
+
+			var src = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+			byteArr := make([]byte, 32) // can be simplified to n/2 if n is always even
+			if _, err := src.Read(byteArr); err != nil {
+				panic(err)
+			}
+
+			randomData := hex.EncodeToString(byteArr)[:64]
+
+			msg := createViewingKeyMsg(randomData)
+
+			return cli.ExecuteWithData(cmd, contractAddr, msg, "", false, "", "", cdc)
+		},
+	}
+
+	return cmd
+}
+
+func s20SetViewingKey(cdc *codec.Codec) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "set-viewing-key [contract address or label]",
+		Short: "sets the viewing key for your account",
+		Long: `This command is useful if you want to manage multiple secret tokens with the same viewing key. WARNING:
+This should only be used to duplicate keys created with the create-viewing-key command, or if you really really know what
+you're doing`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
I'd add a newline before the warning, to make it easier to notice in the `--help` details

Reply:
also, the `Use` field doesn't mention the viewing key in the second argument